controllers/ptpinterface_controller.go

```diff
@@ -674,3 +691,84 @@ func (r *PtpInterfaceReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&starlingxv1.PtpInterface{}).
 		Complete(r)
 }
+
+// Verify whether we have annotation restore-in-progress
+func (r *PtpInterfaceReconciler) checkRestoreInProgress(instance *starlingxv1.PtpInterface) bool {
+	restoreInProgress, ok := instance.Annotations[cloudManager.RestoreInProgress]
+	if ok && restoreInProgress != "" {
+		return true
+	}
+	return false
+}
+
+// Update status and clear annotation
+func (r *PtpInterfaceReconciler) RestorePtpinterfaceStatus(instance *starlingxv1.PtpInterface) error {
```

Comment:
I don't like such a long method spreading in all the controllers, can you refactor it to multiple methods to avoid the nested logical conditions?

Reply:
after removing deploymentScope and strategyRequired from struct the method has been cut short